src/language-js/needs-parens.js

```diff
@@ -227,14 +228,19 @@ function needsParens(path, options) {
     // fallthrough
     case "TSTypeAssertion":
     case "TSAsExpression":
+    case "TSSatisfiesExpression":
     case "LogicalExpression":
       switch (parent.type) {
         case "TSAsExpression":
           // example: foo as unknown as Bar
           return node.type !== "TSAsExpression";
 
+        case "TSSatisfiesExpression":
+          // example: foo satisfies unknown satisfies Bar
+          return node.type !== "TSSatisfiesExpression";
```

Comment:
Maybe mixed should not add parens either

**Prettier pr-13764**
[Playground link](https://deploy-preview-13764--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEAzCEAEBnAhjAS21QLm0wFcoBrKCAdyh3yJLMwCFcAnAHSnRZc5KrQZNhnHv0HNCxUiJp1GmSVz4CMapWNV55bchpAAaEBAAOhaNmSge3BgAUeCOylwAbergCeduYARty4YNRwMADKuAC2cAAyBFBwyKje2HDBoeGRUZZhyQDmyDDcFFkgcLFBcAAmdfUJuFBFFLhFcABiENyx+IStyCC4FDAQZiAAFjCxXgDqUwTw2AVgcFHuywQAbst+w2DYgSDJmdwwzqFF-WkZlQBW2AAeUcVecACKFBDwd16ZcwFbjnYZBXC1LwAWhgJ0s3GSMHmBDqMCmyAAHAAGIFOTLzUKWYbwshwbg7VLmACOP3gVysHhG2ChKXq9Um3DgNIInKuHVuSHSAMqmViBFK5RF7y+tNSgvu5hgEORqPRSAATIrQgQvMUAMIQWICkZeLyTCiZAAqEI8QsBIB2FQAklBGrAomAEdYAIKuqIwPwff6ZAC+IaAA)
<!-- prettier-ignore -->
```sh
--parser babel-ts
```

**Input:**
<!-- prettier-ignore -->
```tsx
foo satisfies unknown satisfies Bar
foo as unknown as Bar
foo satisfies unknown as Bar
foo as unknown satisfies Bar
```

**Output:**
<!-- prettier-ignore -->
```tsx
foo satisfies unknown satisfies Bar;
foo as unknown as Bar;
(foo satisfies unknown) as Bar;
(foo as unknown) satisfies Bar;

```